### DISCUSSION TITLE
Bun 1.0 could yield great benefits for Colyseus

### DISCUSSION THREAD

**Discussion by kragil** [colyseus/colyseus#625] [Ideas]
https://bun.sh/blog/bun-v1.0

Should be fairly easy to get it working with Colyseus, but I failed. 

Maybe more knowledgeable people could give it a try. It would be appreciated.

**Comment by endel** [comment]
Hi @kragil, Bun support would be super. @afrokick also demonstrated interest in this.

These are my notes for Bun so far:

- `internal-ip` module is triggering an exception when loading on bun - This dependency can be easily removed from the @colyseus/core, as it is legacy from [colyseus/proxy](https://github.com/colyseus/proxy) usage and can be easily handled outside the core.
- The default [`@colyseus/ws-transport`](https://github.com/colyseus/colyseus/tree/master/packages/transport/ws-transport) doesn't seem to work out-of-the-box with Bun.
- Creating a bun-specific transport (`@colyseus/bun-websocket`) is probably the way to go!

**Comment by afrokick** [comment]
I definitely will use Bun instead of NodeJS. It'll be nice to have Colyseus + Bun integration as well.

**Reply by kragil** [reply]
Yeah! That was fast! https://github.com/colyseus/colyseus/releases/tag/0.15.8    THANKS!
